dlv attach panics on the first attempt

The scenario is to run GAE application via `dev_appserver.py` script and try to attach to the created process with Dlv:

`C:\Users\Kazz\.IntelliJIdea2018.3\config\plugins\intellij-go\lib\dlv\windows\dlv.exe --listen=localhost:55442 --headless=true --api-version=2 attach 2168 --`

The result is the panic

```
API server listening at: 127.0.0.1:55442
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x68569a]

goroutine 1 [running]:
github.com/derekparker/delve/pkg/proc.(*BinaryInfo).Close(0xc0000b1380, 0x715701, 0xc000060101)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/pkg/proc/bininfo.go:419 +0x4a
github.com/derekparker/delve/pkg/proc/native.(*Process).postExit(0xc00010d320)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/pkg/proc/native/proc.go:410 +0x66
github.com/derekparker/delve/pkg/proc/native.(*Process).Detach(0xc00010d320, 0xc000223e01, 0x29, 0xc0000956b0)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/pkg/proc/native/proc.go:130 +0x20f
github.com/derekparker/delve/pkg/proc/native.Attach(0x878, 0xc0000501b0, 0x1, 0x1, 0x15, 0xc00006e4c0, 0x15)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/pkg/proc/native/proc_windows.go:166 +0xf0
github.com/derekparker/delve/service/debugger.(*Debugger).Attach(0xc0000c4730, 0x878, 0x0, 0x0, 0x1, 0x1, 0x58, 0x58)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/debugger/debugger.go:171 +0x148
github.com/derekparker/delve/service/debugger.New(0xc0000c02a0, 0xc000168470, 0x0, 0x4, 0x0, 0x0, 0x0)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/debugger/debugger.go:96 +0x8d9
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).Run(0xc0000c0240, 0xc0000c0240, 0xc000062360)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/service/rpccommon/server.go:123 +0x130
github.com/derekparker/delve/cmd/dlv/cmds.execute(0x878, 0xc000168470, 0x0, 0x4, 0xc00005a2a0, 0x0, 0x0, 0x3, 0x0)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:587 +0x3e4
github.com/derekparker/delve/cmd/dlv/cmds.attachCmd(0xc00017e480, 0xc000168460, 0x1, 0x5)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:472 +0xc7
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc00017e480, 0xc00005a7e0, 0x5, 0x6, 0xc00017e480, 0xc00005a7e0)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:647 +0x24b
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00017e240, 0x0, 0x0, 0x0)
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:733 +0x2d3
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc00017e240, 0xc00017e240, 0xa2e00f)
C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:692 +0x32
main.main()
    C:/BuildAgent/work/gopath/src/github.com/derekparker/delve/cmd/dlv/main.go:24 +0xe8

Debugger finished with exit code 2
```

The original issue is https://youtrack.jetbrains.com/issue/GO-6798. Feel free to ask details the or here (I'll pass them there)

--- a/pkg/proc/bininfo.go
+++ b/pkg/proc/bininfo.go
@@ -416,7 +416,10 @@ func (bi *BinaryInfo) Close() error {
 	if bi.sepDebugCloser != nil {
 		bi.sepDebugCloser.Close()
 	}
-	return bi.closer.Close()
+	if bi.closer != nil {
+		return bi.closer.Close()
+	}
+	return nil
 }
 
 func (bi *BinaryInfo) setLoadError(fmtstr string, args ...interface{}) {

